### PR TITLE
[d3-delaunay] Update defintions from v4.1 to v5.3

### PR DESCRIPTION
**Pull request by omegafury**
Updated d3-delaunay package types to account for missing and changed method signatures.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/d3/d3-delaunay
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

File: types/d3-delaunay/d3-delaunay-tests.ts [hunk 1]
```diff
@@ -11,24 +11,49 @@ import * as d3 from 'd3-delaunay';
...
+const validContext = {
+    moveTo: (x: number, y: number) => {}, // Delaunay.MoveContext
+    lineTo: (x: number, y: number) => {}, // Delaunay.LineContext
+    closePath: () => {}, // Delaunay.ClosableContext
+    arc: (x: number, y: number, radius: number, startAngle: number, endAngle: number, counterClockwise?: boolean) => {}, // Delaunay.ArcContext
+    rect: (x: number, y: number, width: number, height: number) => {}, // Delaunay.RectContext
+};
+const invalidContext = {
+    moveTo: (x: number, y: number) => {},
+};
```
Comment (omegafury) [inline types/d3-delaunay/d3-delaunay-tests.ts:52]: Added these context objects to test several overloaded methods that were previously only testing the no argument version.

File: types/d3-delaunay/index.d.ts [hunk 2]
```diff
@@ -178,46 +183,6 @@ export namespace Delaunay {
      */
     type GetCoordinate<P, PS> = (point: P, i: number, points: PS) => number;
 
-    /**
```
Comment (omegafury) [inline types/d3-delaunay/index.d.ts:181]: See the "hull" property. It changed from this Node interface to an int32Array, so this interface is no longer needed.

File: types/d3-delaunay/index.d.ts [hunk 3]
```diff
@@ -143,11 +141,18 @@ export class Delaunay<P> {
      */
     renderPoints(context: Delaunay.MoveContext & Delaunay.ArcContext, radius?: number): void;
 
+    /**
+     * Updates the triangulation after the points have been modified in-place.
+     */
+    update(): Delaunay<P>;
```
Comment (omegafury) [inline types/d3-delaunay/index.d.ts:147]: The return types of both the Delaunay.update and Voronoi.update functions aren't abundantly clear in the documentation, but in the source you can see that they each return themselves.

https://github.com/d3/d3-delaunay/blob/64451c23983fce9f63af3a4c36d80a0bec67a234/src/delaunay.js#L47

**Comment by amcasey** [comment]
@Methuselah96 can you please coordinate with @omegafury?

**Comment by Methuselah96** [comment]
@amcasey Yes.

@omegafury I believe https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48208, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48263, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48284, and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48337 make the necessary changes to update d3-delaunay, but feel free to update the PR if I missed something.

**Comment by omegafury** [comment]
@Methuselah96 Taking a look at the diff between our changes right now. I was initially concerned about the type of Delaunay.hull, as the documentation on D3-Delaunay indicates an Int32Array, not a UInt32Array, but on looking at Delaunator where that variable comes from, your type appears to be more correct, and D3-Delaunay's documentation is wrong there.  I'll close this shortly after I finish double checking it.

**Pull request closed without merging**